### PR TITLE
Getting Image-hub functional with Istio

### PR DESCRIPTION
**Pull request by pranav-bhatt**
**Description**
This PR brings updates to the filter so as to provide the user invoking the request, and the number of requests processed so far.
It also revamps the README so as to make deploying using Istio crystal clear.

#### Checklist

- [x] [Signed commits](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin).
- [x] PR title includes `[<name of area>]:<description>`

### CODE REVIEW

**Review by leecalcote — APPROVED** [review]
Looking good. Nice step forward.

**Comment by leecalcote** [comment]
I love seeing your activity here, @pranav-bhatt 👍 

Demo on work-in-progress at tomorrow's community meeting?

**Comment by pranav-bhatt** [comment]
> I love seeing your activity here, @pranav-bhatt +1
> 
> Demo on work-in-progress at tomorrow's community meeting?

Sure!

File: README.md [hunk 1]
```diff
@@ -14,29 +14,65 @@
 Image Hub is a sample application written to run on [Consul](https://meshery.layer5.io/docs/service-meshes/adapters/consul) for exploring WebAssembly modules used as Envoy filters. This demo application has been enabled by experimental works of [Nic Jackson](https://twitter.com/sheriffjackson) of HashiCorp, and [Kanishkar J](https://twitter.com/_kanishkarj_), [Lee Calcote](https://twitter.com/lcalcote), and other contributors of Layer5.
 
 
-### Deployment Instructions (pending [meshery-consul/issues/2](https://github.com/layer5io/meshery-consul/issues/2)):
+## Deployment Instructions:
```
Comment (leecalcote) [inline README.md:17]:
```suggestion
## Deployment Instructions

Image Hub supports Envoy-based data planes. Deployment instructions for each supported service mesh are below.
```

File: rate-limit-filter/src/lib.rs [hunk 2]
```diff
@@ -16,14 +17,31 @@ pub fn _start() {
...
+        .query("SELECT PathName FROM Paths", &[])?
+        .iter()
+        .map(|x| x.get(0))
+        .collect();
+    Ok(array)
+}
+*/
+
+//to be removed
 static ALLOWED_PATHS: [&str; 4] = ["/auth", "/signup", "/upgrade", "/pull"];
```
Comment (leecalcote) [inline rate-limit-filter/src/lib.rs:45]: A to-do item on the list of things to generalize.

File: README.md [hunk 3]
```diff
@@ -14,29 +14,65 @@
...
+### Using Consul:
 
 1) Deploy the latest Consul:
 
-```bash
-helm repo add hashicorp https://helm.releases.hashicorp.com # Adds helm hashicorp repo
-helm install consul hashicorp/consul -f config/consul-values.yaml # Setup custom Consul with support for WASM
-```
+    ```bash
+    helm repo add hashicorp https://helm.releases.hashicorp.com # Adds helm hashicorp repo
```
Comment (leecalcote) [inline README.md:43]: Can we get these changed to deployment instructions using Meshery (the Meshery Adapter for Consul)? // see @mgfeller if help is needed.

File: rate-limit-filter/src/lib.rs (hunk 2 above)
Comment (pranav-bhatt) [reply rate-limit-filter/src/lib.rs:45]: Yup! This PR basically is meant to get the docs and image hub in a state where it's a piece of cake to get going. I've also added headers as @kumarabd requested a couple of days ago, so you should get good insights on the requests as well.

**Pull request merged**